IDE/Renesas/e2studio/RX72N/EnvisionKit/wolfssl_demo/wolfssl_tsip_unit_test.c

```diff
@@ -0,0 +1,1020 @@
+                &g_user_aes128_key_index2);
+                        
+    if (tsip_error_code == TSIP_SUCCESS)
+    tsip_error_code = R_TSIP_GenerateAes256RandomKeyIndex(
+                &g_user_aes256_key_index1);
+    
+    if (tsip_error_code == TSIP_SUCCESS)
+    tsip_error_code = R_TSIP_GenerateAes256RandomKeyIndex(
+                &g_user_aes256_key_index2);
+                            
+    if (tsip_error_code == TSIP_SUCCESS) {
+
+   #ifndef NO_SHA
+        printf(" sha_test()");
+        ret = sha_test();
+        RESULT_STR(ret)
+   #endif
+
+   #ifndef NO_SHA256
+        printf(" sha256_test()");
+        ret = sha256_test();
+        RESULT_STR(ret)
+   #endif
+
+        ret = tsip_aes_cbc_test(1, &g_user_aes128_key_index1);
+
+        if (ret == 0) {
+
+            ret = tsip_aes256_test(1, &g_user_aes256_key_index1);
+
+        }
+
+        if (ret == 0) {
+
+            ret = tsip_aesgcm128_test(1, &g_user_aes128_key_index1);
+
+        }
+        
+        if (ret == 0) {
+
+            ret = tsip_aesgcm256_test(1, &g_user_aes256_key_index1);
+
+        }
+    }
+    else
+        ret = -1;
+
+    return ret;
+}
+
+#ifndef NO_SHA
+
+static void tskSha_Test(void *pvParam)
+{
+    int ret = 0;
+
+    while (exit_loop == false) {
+        ret = sha_test();
+        vTaskDelay(10/portTICK_PERIOD_MS);
+        if (ret != 0) {
+            printf(" result was not good(%d). sha_test\n", ret);
+            sha_multTst_rslt = 1;
+        }
+    }
+
+    xSemaphoreGive(exit_semaph);
+    vTaskDelete(NULL);
+}
+#endif
+
+#ifndef NO_SHA256
+static void tskSha256_Test(void *pvParam)
+{
+    int ret = 0;
+
+    while (exit_loop == false) {
+        ret = sha256_test();
+        vTaskDelay(10/portTICK_PERIOD_MS);
+        if (ret != 0) {
+            printf(" result was not good(%d). sha256_test\n", ret);
+            sha256_multTst_rslt = 1;
+        }
+    }
+
+    xSemaphoreGive(exit_semaph);
+    vTaskDelete(NULL);
+}
+#endif
+
+int tsip_crypt_sha_multitest()
+{
+    int ret = 0;
+    int num = 0;
+    int i;
+    BaseType_t xRet;
+    
+#ifndef NO_SHA
+    num++;
+#endif
+#ifndef NO_SHA256
+    num++;
+#endif
+    exit_loop = 0;
+    sha_multTst_rslt = 0;
+    sha256_multTst_rslt = 0;
+
+    exit_semaph = xSemaphoreCreateCounting(num, 0);
+    xRet = pdPASS;
+    
+#ifndef NO_SHA
+    xRet = xTaskCreate(tskSha_Test, "sha_test", 
+                            SMALL_STACK_SIZE, NULL, 3, NULL);
+#endif
+#ifndef NO_SHA256
+    if (xRet == pdPASS)
+        xRet = xTaskCreate(tskSha256_Test, "sha256_test", 
+                            SMALL_STACK_SIZE, NULL, 3, NULL);
+#endif
+
+    if (xRet == pdPASS) {
+        printf(" Waiting for completing tasks ...");
+        vTaskDelay(10000/portTICK_PERIOD_MS);
+        exit_loop = true;
+
+        for (i = 0; i < num; i++) {
+            if(!xSemaphoreTake(exit_semaph, 2000/portTICK_PERIOD_MS)) {
+                printf("exit semaphore not released by test task");
```

Comment:
In my understanding, `xSemaphoreTake` fails if one of the task is still running and has not called `xSemaphoreGive` yet. If so, this comment is confusing because the task does not take the semaphore. How about saying "test task did not increase the semaphore count".

Reply:
"increase" sounds like mismatch from the API name for me. By using postfix of the API, how about "a semaphore was not given by a test task."?

Reply:
"a semaphore was not given by a test task." sounds good to me.